docs: feature quickstarts and add AG-UI guide (#1229)
google/A2UI: docs/index.md

<div class="grid cards" markdown>

> - :material-clock-fast:{ .lg .middle } **[Restaurant Finder Demo](quickstart.md)**

Reviewer: ```suggestion
- :material-clock-fast:{ .lg .middle } **[Quickstart Restaurant Finder Demo](quickstart.md)**
```

    ---

>     Run the full-stack demo locally with a Gemini agent and Lit renderer. Best for learning the protocol end-to-end.

Reviewer: ```suggestion
    Run the full-stack demo locally with a Gemini powered ADK agent and Lit renderer. Learn A2UI end-to-end and customize to your use case.
```

    [:octicons-arrow-right-24: Run the demo](quickstart.md)

- :material-react:{ .lg .middle } **[A2UI + AG-UI (React)](guides/a2ui-with-any-agent-framework.md)**

    ---

>     Scaffold a Next.js app wired to any agent framework via AG-UI. Best for shipping A2UI in a React application.

Reviewer: ```suggestion
    Scaffold a Next.js app wired to any agent framework via AG-UI. This is a React + A2UI app, ready to ship.
```

    [:octicons-arrow-right-24: Use with any agent](guides/a2ui-with-any-agent-framework.md)

- :material-palette-outline:{ .lg .middle } **[A2UI Composer](https://a2ui-composer.ag-ui.com/)**

    ---

    Generate A2UI JSON from a visual editor — no install required. Paste the output into any agent prompt.

>     [:octicons-arrow-right-24: Open the composer](https://a2ui-composer.ag-ui.com/)

Reviewer: should we also do Theater?
